external-crates/move/move-compiler/src/cfgir/optimize/constant_fold.rs

```diff
@@ -95,9 +95,10 @@ fn optimize_exp(consts: &UniqueMap<ConstantName, Value>, e: &mut Exp) -> bool {
         | E::Unreachable => false,
 
         e_ @ E::Constant(_) => {
-            let name = match e_ {
-                E::Constant(name) => name,
-                _ => unreachable!(),
+            let name = if let E::Constant(name) = e_ {
+                name
+            } else {
+                unreachable!()
             };
```

Comment:
Nit: a perfect use-case for let-else IMO
```suggestion
            let E::Constant(name) = e_ else {
                unreachable!()
            };
```